os/tools/mkbinheader.py

```diff
@@ -116,206 +170,189 @@ def get_static_ram_size(bin_type):
-    if bin_type == BIN :
-        os.system('size ' + elf_path_for_bin_type + ' > ' + STATIC_RAM_ESTIMATION)
-    elif bin_type == ELF :
-        os.system('readelf -S ' + file_path + ' > ' + STATIC_RAM_ESTIMATION)
-    else : #Not supported.
-        print("Error : Not supported Binary Type")
-        sys.exit(1)
-
-    main_priority = int(main_priority)
-    if (main_priority < SCHED_PRIORITY_MIN) or (main_priority > SCHED_PRIORITY_MAX) or (BM_PRIORITY_MIN <= main_priority and main_priority <= BM_PRIORITY_MAX) :
-        print("Error : This binary priority ", main_priority, " is not valid")
-        sys.exit(1)
+    config_path = CONFIG_DIR + '/.config'
+    priority = get_config_value(config_path, "CONFIG_BM_PRIORITY_MAX=")
+    if priority > 0 :
+        BM_PRIORITY_MAX = priority
+    priority = get_config_value(config_path, "CONFIG_BM_PRIORITY_MIN=")
+    if priority > 0 :
+        BM_PRIORITY_MIN = priority
+
+    with open(file_path, 'rb') as fp:
+        # binary data copy to 'data'
+        data = fp.read()
+        file_size = fp.tell()
+        fp.close()
+
+        if binary_format == 'elf' or binary_format == 'ELF' :
+            bin_type = ELF
+        else : # Not supported.
+            bin_type = 0
+            print("Error : Not supported Binary Type")
+            sys.exit(1)
+
+        # Calculate RAM size
+        # Dynamic RAM size : user input of argv[6]
+        # Static RAM size : Extract from size command in linux(ONLY for elf)
+        if bin_type == ELF :
+            os.system('readelf -S ' + file_path + ' > ' + STATIC_RAM_ESTIMATION)
+        else : #Not supported.
+            print("Error : Not supported Binary Type")
+            sys.exit(1)
+
+        main_priority = int(main_priority)
+        if (main_priority < SCHED_PRIORITY_MIN) or (main_priority > SCHED_PRIORITY_MAX) or (BM_PRIORITY_MIN <= main_priority and main_priority <= BM_PRIORITY_MAX) :
+            print("Error : This binary priority ", main_priority, " is not valid")
+            sys.exit(1)
+
+        # Loading priority
+        if loading_priority == 'H' or loading_priority == 'HIGH' :
+            loading_priority = LOADING_HIGH
+        elif loading_priority == 'M' or loading_priority == 'MID' :
+            loading_priority = LOADING_MID
+        elif loading_priority == 'L' or loading_priority == 'LOW' :
+            loading_priority = LOADING_LOW
+        else : #Not supported.
+            print("Error : Not supported Binary loading priority")
+            sys.exit(1)
+
+        static_ram_size = get_static_ram_size(bin_type)
+        cfg_path = CONFIG_DIR + '/.config'
+        if check_optimize_config(cfg_path) == True:
+            binary_ram_size = int(dynamic_ram_size)
+        else:
+            binary_ram_size = int(static_ram_size) + int(dynamic_ram_size)
+            binary_ram_size = roundup_power_two(binary_ram_size)
+
+        if float(kernel_ver) <= 0.0 :
+            kernel_ver = 3.0
```

Comment:
In Kconfig and `make_kernel_binary_header()`, default kernel version is int type and its value is based on `CONFIG_BINARY_VERSION`. But you set the default version as 3.0 in here, even float type.
I think we should unify the version.